### DISCUSSION TITLE
Feature request: Interface for handling images (for the purpose of neovim GUIs)

### DISCUSSION THREAD

**Discussion by Karthik-Dulam** [neovim/neovim#25913] [Ideas]
As goneovim's maintainer [explains here](https://github.com/akiyosi/goneovim/issues/505), a standard interface that all neovim GUI applications can use for image handling is missing and this affects the  development of some features like rendering images in the GUIs.

> One major difficulty is that no standard mechanism or interface for handling images has been established in the current Neovim architecture. This means that if implemented in Goneovim, it would be a Goneovim-specific mechanism, which would not allow for sufficient ecosystem growth.
I believe that image handling needs to be a common mechanism in the Neovim GUI, but we are still in the middle of the road regarding the study of this common mechanism.

I think this would be a very useful addition to the Neovim ecosystem.

**Comment by clason** [comment]
You need to be much more specific about what you mean by "handling images".

**Reply by Karthik-Dulam** [reply]
Thank you for your response. I am not involved in the development of any neovim related projects so I do not know the technical details.

Probably user [goneovim maintainer akiyosi](https://github.com/akiyosi) can clarify what they meant by it. I have requested them to do so.

**Reply by clason** [reply]
They can open their own feature request then and avoid the middle man.
